src/sim/trace-extract/google_trace_simulator.cc

```diff
@@ -445,12 +633,24 @@ void GoogleTraceSimulator::LoadMachineEvents() {
         LOG(ERROR) << "Unexpected structure of machine events on line "
                    << num_line << ": found " << cols.size() << " columns.";
       } else {
+      	uint64_t timestamp = lexical_cast<uint64_t>(cols[0]);
+      	if (timestamp > FLAGS_runtime) {
+      		// only load the events that we need
+      		break;
+      	}
+
         // schema: (timestamp, machine_id, event_type, platform, CPUs, Memory)
+        uint64_t machine_id = lexical_cast<uint64_t>(cols[1]);
+        if (SpookyHash::Hash32(&machine_id, sizeof(machine_id), SEED)
```

Comment:
Add comment. (e.g. Only considering machines that have a hash < proportion). Maybe proportion should be called max_hash.